lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -3,6 +3,190 @@
+     *
+     * @param {@link (AuthenticationParameters:type)}
+     *
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    loginPopup(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented.");
+    }
+
+    /**
+     * Use when you want to obtain an access_token for your API via opening a popup window in the user's browser
+     * @param {@link AuthenticationParameters}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    acquireTokenPopup(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented.");
+    }
+
+    // #region Silent Flow
+
+    /**
+     * Use this function to obtain a token before every call to the API / resource provider
+     *
+     * MSAL return's a cached token when available
+     * Or it send's a request to the STS to obtain a new token using a hidden iframe.
+     *
+     * @param {@link AuthenticationParameters}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     *
+     */
+    acquireTokenSilent(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented."); 
+    }
+
+    // #endregion
+
+    // #region Window Navigation
+
+    /**
+     * @hidden
+     * Used to redirect the browser to the STS authorization endpoint
+     * @param {string} urlNavigate - URL of the authorization endpoint
+     */
+    private navigateWindow(urlNavigate: string): void {
+        throw new Error("Method not implemented.");
+    }
+
+    // #endregion
+
+    // #region Getters and setters
+
+    /**
+     *
+     * Use to get the redirect uri configured in MSAL or null.
+     * Evaluates redirectUri if its a function, otherwise simply returns its value.
+     * @returns {string} redirect URL
+     *
+     */
+    public getRedirectUri(): string {
+        return this.authModule.getRedirectUri();
+    }
+
+    /**
+     * Use to get the post logout redirect uri configured in MSAL or null.
+     * Evaluates postLogoutredirectUri if its a function, otherwise simply returns its value.
+     *
+     * @returns {string} post logout redirect URL
+     */
+    public getPostLogoutRedirectUri(): string {
+        return this.authModule.getPostLogoutRedirectUri();
+    }
+
+    // #endregion
```

Comment:
Can we break this file up? Constructor in one  file / all public APIs only for token acquisition in one file / redirect URIs and callbacks in one?

Reply:
I'm not sure how that helps really, and in terms of code structure it would be difficult to do. I'm happy to take suggestions on ways to do this, but I'm fairly certain this class won't be longer than 500 lines of code. Iframe and popup code will be moved to separate classes, and most of the protocol logic is in the msal common package. @DarylThayil @jasonnutter @jmckennon Would like to hear your thought as well.

Reply:
I'm also of the opinion that if this class is really becoming too large, the solution isn't to split the class definition into different files, but to split the class itself into different classes, which is what I plan to do with the browser specific code (i.e. iframe, popup, etc.)

Reply:
I'd say that yeah, if it's going to be 500 lines, it's likely we'd want so split it up in some way. I don't think I have a strong opinion about how that necessarily happens, but generally sorting ideas into smaller files is good, IMO.

Reply:
That's fine, I will separate them as I add implementation then. For now just focus on API surface and whether we have logical separation between browser and common.

Reply:
My personal opinion is to split it like this 

login-provider.js
token-provider.js

---

```js 
import { acquireTokenSilentProvider } from './providers/token-provider.js';

export default class PublicClientApplication { 
   acquireTokenSilent(request: Request): Promise<Response> {
      //log / telemtry
      // create correlation id? 
      return acquireTokenSilentProvider(request, correlationId)
              .catch(err => {
                    // able to log or transform
                    throw err;
              })
  }
}

This way you separate api / resource from business logic / provider 
The business logic can change but the api stays the same. 
Also promotes reuse

Reply:
I think we should reserve judgement for this until later, when more implementation code is added.